Dockerfile

```diff
@@ -57,13 +60,22 @@ RUN apt-get -y install git
 RUN wget -O cmake.sh https://github.com/Kitware/CMake/releases/download/v3.23.1/cmake-3.23.1-Linux-x86_64.sh && \
     sh ./cmake.sh --prefix=/usr/local --skip-license
 
+
+# Specify that we will use binary LLVM-13 packages
+ENV SYSTEM_LLVM=13
+#Install binary LLVM only if SYTEM_LLVM is not "no"
+RUN if [ "${SYSTEM_LLVM}" != "no" ]; then wget https://apt.llvm.org/llvm.sh && \
+    chmod +x ./llvm.sh && \
+    DEBIAN_FRONTEND=noninteractive ./llvm.sh ${SYSTEM_LLVM} && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -yq libmlir-${SYSTEM_LLVM}-dev; fi
```

Comment:
While this is no problem for docker, I would prefer that LLVM libraries be installed to `$S2EPREFIX/install/ `. I believe this is already done when the first stage of the build downloads and unzips clang+llvm: https://github.com/S2E/s2e/blob/master/Makefile#L294. Wouldn't it be enough to use that instead of installing something else?

Reply:
It should be possible to make it use the binary only LLVM. Currently the Makefile assumes a systemwide installed package.

Reply:
Yes, let's fix that so that it looks for it in `$S2EPREFIX/install/`. Let's also keep the possibility of building LLVM from source. This is particularly useful for debugging low level klee/llvm code.

Reply:
@vitalych So it turns out there's a slight problem with using the binary distribution of LLVM. 

The binary distribution (but not the one distributed via APT, it seems) is compiled without rtti, and exports "-fno-rtti" as a part of it's interface.

Meanwhile, libvmi links against llvm *and* makes use of `dynamic_cast<>`, which requires rtti.

Simply adding "-frtti" to CXX_FLAGS for libvmi does not work -- the final library fails to link since there is no typeinfo defined for LLVM-related classes.

There is some good news: *it just so happens* that libvmi uses `dynamic_cast<>` to deal with a type heirarchy, which is perfect for implementing the plumbing required to get LLVM's `dyn_cast<>` to operate. I could make this fix, but I do not know enough about libvmi to ensure nothing is broken. Would you be able to test it after the change?

Reply:
dynamic_cast is used quite extensively, not just in libvmi. However, it's used on internal classes, not LLVM ones. Would it be possible to mix rtti and non-rtti enabled libraries in that case?

Reply:
I was just looking into this. I was able to get fairly far by judiciously sprinkling in use of `-fno-rtti` on specific source files and minor patches. The stopper for now is `revgen64` -- it uses protobuf (which requires RTTI) and also links to LLVM.

Reply:
So I think there's really two ways to proceed:
1) Build our own LLVM-13 with RTTI enabled and host it somewhere. This can then be used to provide the clang binary and also as a binary LLVM to link against.
2) Use the apt-packaged LLVM-13 if present, build from source otherwise. This is the current behavior, which maybe should include some specific checks for RTTI.

Reply:
It's a bit strange that rtti would contaminate libraries that don't use it. I don't know the details about how it works though, maybe it's the case that merely having one library that uses it is enough to break everything else.

I like solution 1, it's easy to host files on github. Hopefully LLVM has the source of their build system that we could use to generate packages.

Regarding LLVM version, I've just upgraded to 14 and perhaps soon I'll upgrade to 15. I do not support older versions of LLVM, so we'd have to keep installing new deb packages in the system, which I'd rather avoid.

Reply:
Speaking of revgen, I now remember it has some requirements about LTO, which IIRC is not available in the official packages. So building our own is probably indeed the best.